src/v/cloud_storage/topic_manifest_downloader.h

```diff
@@ -57,6 +58,19 @@ class topic_manifest_downloader {
       model::timestamp_clock::duration backoff,
       topic_manifest*);
 
+    // Attempts to scan the topic manifest root paths for any topic manifests
+    // that match the given filter.
+    using tp_ns_filter_t = std::function<bool(const model::topic_namespace&)>;
+    static ss::future<result<find_topic_manifest_outcome, error_outcome>>
+    find_manifests(
+      remote& remote,
+      cloud_storage_clients::bucket_name bucket,
+      retry_chain_node& parent_retry,
+      ss::lowres_clock::time_point deadline,
+      model::timestamp_clock::duration backoff,
+      std::optional<tp_ns_filter_t> tp_filter,
+      chunked_vector<topic_manifest>*);
```

Comment:
chunked_vector<topic_manifest>*

is that output variable something that is built up iteratively, or could it be part of the return value?

Reply:
It could be built up iteratively, but isn't today. At least in a few places recently I've been trying a style where return types are a status/enum, and everything else is an out-parameter. I'm finding it much easier to express expected outcomes that way.